python/tests/unit/test_fetch.py

```diff
@@ -309,33 +309,74 @@ def test_ingress_with_named_port(self):
   rules:
   - http:
       paths:
-      - backend:
+      - path: /
+        pathType: ImplementationSpecific
+        backend:
           serviceName: quote
           servicePort: http
-        path: /
+      - path: /metrics
+        pathType: ImplementationSpecific
+        backend:
+          serviceName: quote
+          servicePort: metrics
+      - path: /health
+        pathType: ImplementationSpecific
+        backend:
+          serviceName: quote
+          servicePort: 9000
+      - path: /missed-name
         pathType: ImplementationSpecific
+        backend:
+          serviceName: missed
+          servicePort: missed
+      - path: /missed-number
+        pathType: ImplementationSpecific
+        backend:
+          serviceName: missed
+          servicePort: 8080
 status:
   loadBalancer: {}
 """
         aconf = Config()
+        logger.setLevel(logging.DEBUG)
+
         fetcher = ResourceFetcher(logger, aconf)
         fetcher.parse_yaml(yaml, True)
 
         mgr = fetcher.manager
-        assert len(mgr.elements) == 2
+        assert len(mgr.elements) == 6
 
         aconf.load_all(fetcher.sorted())
         assert len(aconf.errors) == 0
 
         mappings = aconf.get_config("mappings")
         assert mappings
-        assert len(mappings) == 1
-
-        mapping = next(iter(mappings.values()))
-        assert mapping.name == "quote-0-0"
-        assert mapping.namespace == "default"
-        assert mapping.prefix == "/"
-        assert mapping.service == "quote.default:3000"
+        assert len(mappings) == 5
+
+        mapping_root = mappings.get("quote-0-0")
+        assert mapping_root
+        assert mapping_root.prefix == "/"
+        assert mapping_root.service == "quote.default:3000"
+
+        mapping_metrics = mappings.get("quote-0-1")
+        assert mapping_metrics
+        assert mapping_metrics.prefix == "/metrics"
+        assert mapping_metrics.service == "quote.default:metrics"
+
+        mapping_health = mappings.get("quote-0-2")
+        assert mapping_health
+        assert mapping_health.prefix == "/health"
+        assert mapping_health.service == "quote.default:9000"
+
+        mapping_missed_name = mappings.get("quote-0-3")
+        assert mapping_missed_name
+        assert mapping_missed_name.prefix == "/missed-name"
+        assert mapping_missed_name.service == "missed.default:missed"
```

Comment:
maybe I'm missing something but aren't we trying to resolve the names to port numbers here? 

I would expect this one to be rejected and dropped or fallback to something, same with the metrics above too.

Reply:
Hello, @LanceEa! 
The fallback behavior is to retain the named port in the `aconf` mapping structure (this is how the ingress fetcher works at the moment). The bad port will be replaced to 80/443 later here: https://github.com/emissary-ingress/emissary/blob/aee9d485917365a2efa53ce90f121483da5c95a0/python/ambassador/ir/ircluster.py#L212-L215

Reply:
The point of these additional ingress fetcher tests is to demonstrate that port number resolution works whenever it is possible: referenced service and ports exist. In case of ingress' paths reference non-existing services or ports fetcher behaves as before.
Unfortunately, there were no tests for ingress fetcher before, so there's nothing to compare these tests with.
One thing I can do to better emphasize that the port number resolution is not guaranteed is to add the word `try` to the name of the method.

Reply:
ahh got it. Like I mentioned, I wasn't as familiar with this part of the codebase 😄. My initial gut instinct was to assume that the resolution would have happened at this stage rather than delaying until later stages. So, LGTM